[Profiler][Trivial] Force Result to be a shared_ptr

robieta opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #77697
* #77696
* __->__ #77695
* #77694

A lot of the graph manipulation in later changes will rely on the ability to hold stable references, both in C++ and Python.

Differential Revision: [D36302564](https://our.internmc.facebook.com/intern/diff/D36302564/)

Commit:
[Profiler][Trivial] Force Result to be a shared_ptr

A lot of the graph manipulation in later changes will rely on the ability to hold stable references, both in C++ and Python.

Differential Revision: [D36302564](https://our.internmc.facebook.com/intern/diff/D36302564/)

[ghstack-poisoned]

aaronenyeshi reviewed (approved):
Reviewed internally, LGTM!

Commit:
Update on "[Profiler][Trivial] Force Result to be a shared_ptr"


A lot of the graph manipulation in later changes will rely on the ability to hold stable references, both in C++ and Python.

Differential Revision: [D36302564](https://our.internmc.facebook.com/intern/diff/D36302564/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Trivial] Force Result to be a shared_ptr"


A lot of the graph manipulation in later changes will rely on the ability to hold stable references, both in C++ and Python.

Differential Revision: [D36302564](https://our.internmc.facebook.com/intern/diff/D36302564/)

[ghstack-poisoned]
